src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
+            String groesse = f.arr[DatenFilm.FILM_GROESSE];
+            if (groesse.isEmpty())
+                return false;
+            else {
+                return Long.parseLong(groesse) < MIN_SIZE_ADD_OLD;
+            }
 
-        int count = 0;
-        final int MAX_THREAD = 30;
-        final int MAX_WAIT_TIME = 450; // 450*2=900s -> 15 Minuten
-        //final int MAX_WAIT_TIME = 10; // 10*2=20s
-        stopOld = false;
-        size = listeEinsortieren.size();
+        });
+    }
+
+    private void startThreads() {
+        final OkHttpClient client = MVHttpClient.getInstance().getReducedTimeOutClient();
+        for (int i = 0; i < NUMBER_OF_THREADS; ++i) {
+            ImportOldFilmlistThread t = new ImportOldFilmlistThread(listeEinsortieren, client);
+            t.setName("ImportOldFilmlistThread Thread-" + i);
+            threadList.add(t);
+            t.start();
+        }
+    }
 
-        // Rest nehmen wir wenn noch online
-        for (int i = 0; i < MAX_THREAD; ++i) {
-            new Thread(new AddOld(listeEinsortieren)).start();
+    private void stopThreads() {
+        if (Config.getStop()) {
+            for (ImportOldFilmlistThread t : threadList)
+                t.interrupt();
+            for (ImportOldFilmlistThread t : threadList) {
+                try {
+                    t.join();
+                } catch (InterruptedException ignored) {
+                }
+            }
         }
+    }
+    
+    /*
+     * Diese Methode sortiert eine vorhandene Liste in eine andere Filmliste ein, 
+     * dabei werden nur nicht vorhandene Filme einsortiert.
+     */
+    public int addOldList() {
+        threadCounter = new AtomicInteger(0);
+        final HashSet<Hash> hash = new HashSet<>(vonListe.size() + 1);
+
+        performInitialCleanup();
 
+        int size = listeEinsortieren.size();
+
+        performTitleSearch(hash, size);
+
+        size = listeEinsortieren.size();
+        performUrlSearch(hash, size);
+
+        size = listeEinsortieren.size();
+        long oldSize = size;
+
+        startThreads();
+
+        int count = 0;
         while (!Config.getStop() && threadCounter.get() > 0) {
             try {
-                System.out.println("sek.: " + 2 * (count++) + "  Liste: " + listeEinsortieren.size() + "  Treffer: " + treffer.get() + "   Threads: " + threadCounter);
-                if (count > MAX_WAIT_TIME) {
-                    // dann haben wir mehr als 10 Minuten und: Stop
-                    Log.sysLog("===== Liste einsortieren: ABBRUCH =====");
-                    Log.sysLog("COUNT_MAX erreicht [s]: " + MAX_WAIT_TIME * 2);
-                    Log.sysLog("");
-                    stopOld = true;
+                count++;
+                if (count % 5 == 0) {
+                    long curSize = listeEinsortieren.size();
+                    System.out.println("Liste: " + curSize);
+                    System.out.println("Entfernte Einträge: " + ((oldSize - curSize)));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove those useless parentheses. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessParenthesesCheck)